src/marshmallow/fields.py

```diff
@@ -537,7 +553,19 @@ def _deserialize(self, value, attr, data, partial=None, **kwargs):
         return self._load(value, data, partial=partial)
 
 
-class List(Field):
+class ContainerMixin(object):
+    """Common methods for container fields"""
+
+    def get_container_modifiers(self, container):
+        for attr, value in iteritems(container.get_modifiers()):
+            setattr(self, attr, value)
+
+    def set_container_modifiers(self, container):
+        container.set_modifiers({
+            attr: getattr(self, attr) for attr in CONTAINER_MODIFIERS})
+
+
+class List(Field, ContainerMixin):
```

Comment:
Since Python's MRO goes from left to right, `ContainerMixin` should come before `Field`.